### ISSUE TITLE
Development mode is incorrect in vite dev

### ISSUE THREAD

**Issue by hyunbinseo** [vercel/analytics#25]
## Summary

In a browser accessing a Vite dev server, the value of [`isDevelopment()`](https://github.com/vercel/analytics/blob/74c2474ebbb75c50a0009b067e25c56d5255c6a5/packages/web/src/utils.ts#L5-L10) is `false`.

Therefore `/_vercel/insights/script.js` is requested, resulting in a `404 Not Found`.

## Reason

In a browser accessing a Vite dev server, the `process` is `undefined`.

https://github.com/vercel/analytics/blob/74c2474ebbb75c50a0009b067e25c56d5255c6a5/packages/web/src/utils.ts#L5-L10

Therefore `isDevelopment()` returns `false`.

## Reproduction

Open the browser console to see what is logged in the client inside [StackBlitz](https://stackblitz.com/).

- [SvelteKit (Vite)](https://stackblitz.com/edit/sveltejs-kit-template-default-kwvuew?file=src/routes/+layout.svelte)
- [Next.js](https://stackblitz.com/edit/nextjs-x9hb9b?file=pages/_app.js)

## Values

In a browser accessing a Vite dev server:

|                             | SvelteKit    | Next.js     |
| --------------------------- | ------------ | ----------- |
| isBrowser()                 | true         | true        |
| isDevelopment()             | false \*     | true        |
| typeof process              | undefined \* | object      |
| typeof process.env.NODE_ENV | string       | string      |
| process.env.NODE_ENV        | development  | development |

**Comment by hyunbinseo** [comment]
Ability to provide a custom `isBrowser` and `isDevelopment` value would be a cross-framework solution.

For example, [SvelteKit](https://kit.svelte.dev/docs/modules#$app-environment) provides both of them:

```ts
import { browser, dev } from '$app/environment';
```

**Comment by hyunbinseo** [comment]
Thank you for the quick fix. It works as expected.

```ts
import { browser, dev } from '$app/environment';
import { inject } from '@vercel/analytics';

inject({ __mode: dev ? 'development' : 'production' });
```

In a browser accessing a Vite dev server, the following message is logged.

```
[Vercel Analytics] Debug mode is enabled by default in development. No requests will be sent to the server.
```

---

I hope there is an option to override `is-browser` and `is-development`.

For example, [SvelteKit](https://kit.svelte.dev/docs/modules#$app-environment) provides both `browser` and `dev` boolean values.

```ts
import { browser, dev } from '$app/environment';
import { inject } from '@vercel/analytics';

inject({ browser, dev }); // example
```

If values are not explicitly provided, the library can decide on its own.

**Comment by chriswdmr** [comment]
Awesome!
Side note: we're going to ship it as `mode` instead of `__mode`.

You can disable those log messages by providing the `debug` property with `false` if your mode is `development`.

Please [see our documentation](https://vercel.com/docs/concepts/analytics/audiences/package#debug) around the `debug` property

**Comment by hyunbinseo** [comment]
@chriswdmr [v0.1.6](https://github.com/vercel/analytics/releases/tag/0.1.6) looks great.

Should this issue be closed? `mode` still seems to be a workaround.

If all Vite projects still require the `mode` property to be explicitly set,

1. It should be clearly stated in the [documentation](https://vercel.com/docs/concepts/analytics/audiences/package)'s 'Other Frameworks' section
2. or, the `isDevelopment` check should be improved to support Vite out of the box.

**Comment by chriswdmr** [comment]
Perfect, thanks for confirmation.

Closing as it's resolved with [v0.1.6](https://github.com/vercel/analytics/releases/tag/0.1.6)

Documentation is already updated (will be shipped in ~1 hour)
